File: src/scala_answer/Lessons12939.scala
package scala_answer

/**
  * @author dave.th
  * https://programmers.co.kr/learn/courses/30/lessons/12939
  */
object Lessons12939 {
  def main(args: Array[String]) : Unit = {
    println(solution("-1 -1"))
  }

  def solution(s: String) : String = {
    val array = s.split(" ").map(i => i.toInt)
    array.min + " " + array.max
  }
}

File: src/java_answer/Lessons12939.java
package java_answer;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

import static java.lang.System.out;

/**
 * @author dave.th
 * https://programmers.co.kr/learn/courses/30/lessons/12939
 */
public class Lessons12939 {
    public static void main(String[] args) {
        out.println((new Lessons12939()).solution("-1 -4"));
    }

    private String solution(String s) {
        List<Integer> list = Arrays.stream(s.split(" "))
                .map(Integer::parseInt)
                .collect(Collectors.toList());

        int min = list.stream().min(Integer::compareTo).orElse(0);
        int max = list.stream().max(Integer::compareTo).orElse(0);

        return min + " " + max;
    }
}
